File: Frontend/ParkingApp/src/app/models/ParkingLotStatus.ts
/**
 * Parking lot statuses
 */
export const parkingStatuses = {
    FREE: 'FREE',
    OCCUPIED: 'OCCUPIED',
    UNKNOWN: 'UNKNOWN',
    RESERVED: 'RESERVED'
};

/**
 * Parking lot statuses colors
 */
export const parkingColors = [
    { status: parkingStatuses.FREE, background: '#28a745' },
    { status: parkingStatuses.OCCUPIED, background: '#dc3545' },
    { status: parkingStatuses.UNKNOWN, background: 'gray' },
    { status: parkingStatuses.RESERVED, background: '#ffbf0f' }
];

File: Backend/src/main/java/com/isd/parking/models/enums/ParkingLotStatus.java
package com.isd.parking.models.enums;

import java.util.List;
import java.util.Random;


/**
 * Permissible parking lot statuses enumeration
 */
public enum ParkingLotStatus {
    FREE,
    OCCUPIED,
    RESERVED,
    UNKNOWN;

    /**
     * Gets random parking lot status
     * Used for fake statistics generator
     */
    private static final List<ParkingLotStatus> VALUES = List.of(values());
    private static final int SIZE = VALUES.size();
    private static final Random RANDOM = new Random();

    public static ParkingLotStatus getRandomStatus()  {
        return VALUES.get(RANDOM.nextInt(SIZE));
    }
}
